druid/src/widget/scroll.rs

```diff
@@ -162,13 +240,21 @@ impl<T, W: Widget<T>> Scroll<T, W> {
     /// Update the scroll.
     ///
     /// Returns `true` if the scroll has been updated.
-    pub fn scroll(&mut self, delta: Vec2, size: Size) -> bool {
-        let mut offset = self.scroll_offset + delta;
-        offset.x = offset.x.min(self.child_size.width - size.width).max(0.0);
-        offset.y = offset.y.min(self.child_size.height - size.height).max(0.0);
+    pub fn scroll_by(&mut self, submit: &mut SubmitCommand, delta: Vec2, size: Size) -> bool {
```

Comment:
I see you've decided to go for the breaking change. Which is fine, but it (along with the other publicly-visible changes in this PR) will need a CHANGELOG entry...

Reply:
Ok. My thoughts were that the signature had changed so much, may as well break the name.
It already was slightly odd, due to the size, which was obtained from a context from the scrolls own widget pod - calling it from outside meant doing tricks to obtain that. So it was never great as public api - what would be needed for that is “scroll here when you can”, ie a pending scroll - but this is recreating the command system afaict. I’m going to make an alternate PR, based on the binding/scope stuff I mentioned on Zulip. It’s much closer to the composition you are talking about above, and there aren’t any callbacks. Then people can pick between approaches.